cmd/kubeadm/app/preflight/checks_windows.go

```diff
@@ -20,34 +20,20 @@ limitations under the License.
 package preflight
 
 import (
-	"os/user"
-
 	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
 )
 
-// The "Well-known SID" of Administrator group
-// https://support.microsoft.com/en-us/help/243330/well-known-security-identifiers-in-windows-operating-systems
-const administratorSID = "S-1-5-32-544"
-
 // Check validates if a user has elevated (administrator) privileges.
 func (ipuc IsPrivilegedUserCheck) Check() (warnings, errorList []error) {
-	currUser, err := user.Current()
-	if err != nil {
-		return nil, []error{errors.Wrap(err, "cannot get current user")}
-	}
-
-	groupIds, err := currUser.GroupIds()
-	if err != nil {
-		return nil, []error{errors.Wrap(err, "cannot get group IDs for current user")}
+	var hProcessToken windows.Token
+	if err := windows.OpenProcessToken(windows.CurrentProcess(), windows.TOKEN_QUERY, &hProcessToken); err != nil {
```

Comment:
Might be good idea to call  `token.close` on this token or use https://pkg.go.dev/golang.org/x/sys/windows#GetCurrentProcessToken which states it doesn't need to be closed.

Reply:
great, i didn't notice that function earlier.
updated.